File: plugin/src/org/jetbrains/yesod/hamlet/parser/HamletTokenTypes.java
package org.jetbrains.yesod.hamlet.parser;

/**
 * @author Leyla H
 */

import com.intellij.psi.TokenType;
import com.intellij.psi.tree.IElementType;
import com.intellij.psi.tree.TokenSet;

public interface HamletTokenTypes {

    IElementType COMMENT_END = new HamletToken("-->");
    IElementType BINDSTATMENT = new HamletToken("<-");
    IElementType COMMENT_START = new HamletToken("<!--");
    IElementType BACKSLASH = new HamletToken("\\");
    IElementType DOLLAR = new HamletToken("$");
    IElementType SLASH = new HamletToken("/");
    IElementType DOT = new HamletToken(".");
    IElementType END_INTERPOLATION = new HamletToken("}");
    IElementType EQUAL = new HamletToken("=");
    IElementType OANGLE = new HamletToken("<");
    IElementType CANGLE = new HamletToken(">");

    IElementType IDENTIFIER = new HamletToken("identifier");
    IElementType DOT_IDENTIFIER = new HamletToken("dot_identifier");
    IElementType COLON_IDENTIFIER = new HamletToken("colon_identifier");
    IElementType SHARP_IDENTIFIER = new HamletToken("sharp_identifier");
    IElementType OPERATOR = new HamletToken("operator");
    IElementType STRING = new HamletToken("string");

    IElementType NEWLINE = new HamletToken("newline");
    IElementType DOCTYPE = new HamletToken("doctype");
    IElementType COMMENT = new HamletToken("comment");
    IElementType INTERPOLATION = new HamletToken("interpolation");

    IElementType ANY = new HamletCompositeElementType("ANY");
    IElementType TAG = new HamletCompositeElementType("TAG");
    IElementType ATTRIBUTE = new HamletCompositeElementType("ATTRIBUTE");
    IElementType ATTRIBUTE_VALUE = new HamletCompositeElementType("ATTRIBUTE_VALUE");

    TokenSet WHITESPACES = TokenSet.create(TokenType.WHITE_SPACE);

}

File: plugin/src/org/jetbrains/yesod/hamlet/parser/HamletTokenTypes.kt
package org.jetbrains.yesod.hamlet.parser

/**
 * @author Leyla H
 */

import com.intellij.psi.TokenType
import com.intellij.psi.tree.IElementType
import com.intellij.psi.tree.TokenSet

object HamletTokenTypes {
    @JvmField
    val COMMENT_END: IElementType = HamletToken("-->")

    @JvmField
    val BINDSTATMENT: IElementType = HamletToken("<-")

    @JvmField
    val COMMENT_START: IElementType = HamletToken("<!--")

    @JvmField
    val SLASH: IElementType = HamletToken("/")

    @JvmField
    val DOT: IElementType = HamletToken(".")

    @JvmField
    val END_INTERPOLATION: IElementType = HamletToken("}")

    @JvmField
    val EQUAL: IElementType = HamletToken("=")

    @JvmField
    val OANGLE: IElementType = HamletToken("<")

    @JvmField
    val CANGLE: IElementType = HamletToken(">")

    @JvmField
    val IDENTIFIER: IElementType = HamletToken("identifier")

    @JvmField
    val DOT_IDENTIFIER: IElementType = HamletToken("dot_identifier")

    @JvmField
    val COLON_IDENTIFIER: IElementType = HamletToken("colon_identifier")

    @JvmField
    val SHARP_IDENTIFIER: IElementType = HamletToken("sharp_identifier")

    @JvmField
    val OPERATOR: IElementType = HamletToken("operator")

    @JvmField
    val STRING: IElementType = HamletToken("string")

    @JvmField
    val ESCAPE: IElementType = HamletToken("escape")

    @JvmField
    val NEWLINE: IElementType = HamletToken("newline")

    @JvmField
    val DOCTYPE: IElementType = HamletToken("doctype")

    @JvmField
    val COMMENT: IElementType = HamletToken("comment")

    @JvmField
    val INTERPOLATION: IElementType = HamletToken("interpolation")

    @JvmField
    val ANY: IElementType = HamletCompositeElementType("ANY")

    @JvmField
    val TAG: IElementType = HamletCompositeElementType("TAG")

    @JvmField
    val ATTRIBUTE: IElementType = HamletCompositeElementType("ATTRIBUTE")

    @JvmField
    val ATTRIBUTE_VALUE: IElementType = HamletCompositeElementType("ATTRIBUTE_VALUE")

    @JvmField
    val WHITESPACES: TokenSet = TokenSet.create(TokenType.WHITE_SPACE)

}
